lib/reporting/frontend_data.rb

```diff
@@ -44,12 +44,20 @@ def order_cycles
         order('order_cycles.orders_close_at DESC')
     end
 
+    def order_customers
+      Customer.where(id: visible_order_customer_ids).select("customers.id, customers.email")
+    end
+
     private
 
     attr_reader :current_user
 
     def permissions
       @permissions ||= OpenFoodNetwork::Permissions.new(current_user)
     end
+
+    def visible_order_customer_ids
+      Permissions::Order.new(current_user).visible_orders.pluck(:customer_id)
```

Comment:
Why did you choose `pluck` over `select` here?

Reply:
because `pluck` is faster than `select`, it doesn't create an active record instance.

[benchmark](https://phrase.com/blog/posts/activerecord-speed-up-your-sql-queries/)

Reply:
Oh, I thought it's used within another query and therefore just a sub-query within the database instead of going though Rails.